### PR TITLE
[3201] Add support for tools on multiple diagram elements (hide, fade…

### LINKED ISSUE #3201: Add support for frontend tools on multiple diagram elements
Supported tools: 
- hide
- fade
- pin

### PR DESCRIPTION
**Pull request by frouene**
… and pin)

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3201

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  if (!shouldRender) {
+    return null;
+  }
+  hideDiagramElementPalette();
+
+  return (
+    <PalettePortal>
+      <Paper className={classes.palette} style={{ position: 'absolute', left: x, top: y }} data-testid="GroupPalette">
+        <div className={classes.paletteContent}>
+          <Tooltip title="Hide elements">
```
Comment (AxelRICHARD) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx:83]: I see in `Palette.tsx` that `Hide elements` is also used as label for this action, but it should be `Hide element`.
Could you fix that in this PR please and make sure the cypress tests are also updated accordingly?
Also do the same for `Fade`.

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            <IconButton
+              className={classes.toolIcon}
+              size="small"
+              aria-label="Fade elements"
+              onClick={() => fadeDiagramElements(selectedElementIds, true)}
+              data-testid="Fade-elements">
+              <TonalityIcon fontSize="small" />
+            </IconButton>
+          </Tooltip>
+          <Tooltip title="Pin element">
```
Comment (AxelRICHARD) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx]: Could you please rename into `Pin elements` please? (also change aria-label and data-testid)

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -130,7 +130,7 @@ image:doc/screenshots/showDeletionConfirmation.png[Deletion Confirmation Dialog,
 - https://github.com/eclipse-sirius/sirius-web/issues/3193[#3193] [core] Added the methods `hasVariable(String name)` and `getParent()` in VariableManager to, respectively, detect if a variable manager defines a variable (it returns `false`, if the variable is defined by a parent `VariableManager`) and to navigate to a parent `VariableManager`.
 Thanks to this capability, a specifier can easily retrieve a variable overriden by a child `VariableManager`.
 - https://github.com/eclipse-sirius/sirius-web/issues/3096[#3096] [tree] Tree representations can now support filters
-
+- https://github.com/eclipse-sirius/sirius-web/issues/3201[#3201] [diagram] Add support for tools on multiple diagram elements (hide, fade and pin).
```
Comment (AxelRICHARD) [inline CHANGELOG.adoc]: Add support for tools on multiple diagram elements selection (hide, fade and pin).

File: CHANGELOG.adoc
Comment (AxelRICHARD) [inline CHANGELOG.adoc:1]: when 3 or more elements are selected, `Escape` key doesn't close the palette. could you fix that please?

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx (hunk 1 above)
Comment (frouene) [reply packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx:83]: done

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx (hunk 2 above)
Comment (frouene) [reply packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx]: done

File: CHANGELOG.adoc (hunk 3 above)
Comment (frouene) [reply CHANGELOG.adoc]: done

File: CHANGELOG.adoc
Comment (frouene) [reply CHANGELOG.adoc:1]: done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**